bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/handler/EnergyManagerHandler.java

```diff
@@ -0,0 +1,397 @@
+            return localDevices.get(guid);
+        }
+
+        return null;
+    }
+
+    /**
+     * Convert the energy manager millisecond timestamps to {@link ZonedDateTime}
+     *
+     * The energy manager is the only point that knows about the timezone and
+     * it is available to all other devices. All timestamps used by all devices
+     * are in milliseconds since the epoch.
+     *
+     * @param timestamp milliseconds since the epoch
+     * @return date time in timezone
+     */
+    public ZonedDateTime getFromMilliTimestamp(BigDecimal timestamp) {
+        Map<String, Device> devices = this.getDevices();
+        if (devices != null) {
+            EnergyManager energyManager = (EnergyManager) devices.get(this.energyManagerGuid);
+            if (energyManager != null) {
+                BigDecimal[] bigDecimals = timestamp.divideAndRemainder(BigDecimal.valueOf(1_000));
+                Instant instant = Instant.ofEpochSecond(bigDecimals[0].longValue(),
+                        bigDecimals[1].multiply(BigDecimal.valueOf(1_000_000)).longValue());
+
+                ZoneId localZoneID = this.zoneId;
+                if (localZoneID != null) {
+                    return ZonedDateTime.ofInstant(instant, localZoneID);
+                }
+            }
+        }
+
+        throw new DateTimeException("Timezone from energy manager missing.");
+    }
+
+    /**
+     * Reload all devices from the energy manager.
+     *
+     * This method is called via the {@link ExpiringCache}.
+     * 
+     * @return map from guid to {@link Device}}
+     */
+    private @Nullable Map<String, Device> refreshDevices() {
+        try {
+            final Map<String, Device> devicesData = this.connector.retrieveDevices().getDevices();
+            this.updateStatus(ThingStatus.ONLINE);
+
+            // trigger refresh of the available channels
+            if (devicesData.containsKey(this.energyManagerGuid)) {
+                Device device = devicesData.get(this.energyManagerGuid);
+                if (device != null) {
+                    this.initDeviceChannels(device);
+                }
+            } else {
+                this.logger.warn("{}: initDeviceChannels missing energy manager {}", this, this.getThing().getUID());
+            }
+
+            return devicesData;
+        } catch (final SolarwattConnectionException e) {
+            this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+
+        return null;
+    }
+
+    /**
+     * Trigger an update on all child things of this bridge.
+     */
+    private void updateAllChildThings() {
+        this.getThing().getThings().forEach(childThing -> {
+            try {
+                ThingHandler childHandler = childThing.getHandler();
+                if (childHandler != null) {
+                    childHandler.handleCommand(new ChannelUID(childThing.getUID(), "guid"), RefreshType.REFRESH);
```

Comment:
If I see correctly, there is no Channel called "guid". You could use e.g. "timestamp".

Reply:
The handleCommand in SimpleDeviceHandler always updates everything as there is no way to just get a single value from the energy-manager. So the given channel is effectively ignored.
```
    public void handleCommand(ChannelUID channelUID, Command command) {
        final EnergyManagerHandler bridgeHandler = this.getEnergyManagerHandler();
        if (bridgeHandler != null) {
            if (command instanceof RefreshType) {
                this.updateDeviceProperties();
                this.updateDeviceChannels();
            }
        } else {
            this.logger.warn("Thing {} has no bridgeHandler for Bridge {}", this.getThing().getUID(),
                    this.getThing().getBridgeUID());
        }
    }
```

Reply:
I see. But it feels wrong to use a non-existent channel name, which is even a config parameter name. Why not using some channel that exists?

Reply:
I added idName as a channel for all devices and now use that for the refresh.

Reply:
It sounds `idName` is rather a configuration parameter or a property. So, it shouldn't be a Channel. The bridge has some existing Channels, why not using some of them for the refresh?

Reply:
I changed it to use your first suggestion. It now uses "timestamp" as channel.